soccer/src/soccer/strategy/agent/position/runner.cpp

```diff
@@ -0,0 +1,266 @@
+#include "runner.hpp"
+
+namespace strategy {
+
+Runner::Runner(int r_id) : Position(r_id) {
+    position_name_ = "Runner";
+    current_state_ = IDLING; 
+}
+
+std::optional<RobotIntent> Runner::derived_get_task(RobotIntent intent) {
+    current_state_ = update_state();
+    return state_to_task(intent);
+}
+
+Runner::State Runner::update_state() {
+    State next_state = current_state_;
+    // handle transitions between current state
+    WorldState* world_state = this->world_state();
+
+    //SPDLOG_INFO("ANSON: test test test");
+
+    // if no ball found, stop and return to box immediately
+    if (!world_state->ball.visible) {
+        return current_state_;
+    }
+
+    rj_geometry::Point robot_position = world_state->get_robot(true, robot_id_).pose.position();
+    rj_geometry::Point ball_position = world_state->ball.position;
+    double distance_to_ball = robot_position.dist_to(ball_position);
+
+    if (current_state_ == IDLING) {
+        //send_scorer_request();
+        next_state = MOVING_LSIDE;
+    } else if (current_state_ == MOVING_LSIDE && at_corner) {
```

Comment:
something like `at_corner` should be calculated in `update_state()` (since that is part of your state transition) and *not* in `state_to_task`, which just converts states to tasks.

Reply:
However, for a simple position like `Runner`, we shouldn't be manually calculating things like `at_corner`. Use the provided `check_is_done()` function to check if the previous motion command has completed.